cantools/database/can/c_source.py

```diff
@@ -668,10 +663,10 @@ def _format_decode_code_signal(message,
     if signal.is_float:
         if signal.length == 32:
             variable = '    uint32_t {} = 0;'.format(signal_name)
-            line = '    dst_p->{0} = itof({0});'.format(signal_name)
+            line = '    memcpy(&dst_p->{0}, &{0}, sizeof({0}));'.format(signal_name)
         else:
             variable = '    uint64_t {} = 0;'.format(signal_name)
-            line = '    dst_p->{0} = itod({0});'.format(signal_name)
+            line = '    memcpy(&dst_p->{0}, &{0}, sizeof({0}));'.format(signal_name)
```

Comment:
I suggest changing to `memcpy(&dst_p->{0}, &{0}, sizeof(dst_p->{0}));` to ensure not writing outside the destination buffer.